### ISSUE TITLE
Bug: Partitions files not deleted when you delete a service

### ISSUE THREAD

**Issue by clemthi** [ferdium/ferdium-app#646]
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

6.2.0 Portable

### What Operating System are you using?

Windows

### Operating System Version

Windows 11

### What arch are you using?

x64

### Expected Behavior

When deleting a service in Ferdium, it would be expected to have the related files to be deleted.
This is not currently the case.

### Actual Behavior

When a service is delete, its folder (located under `.\FerdiumAppData\FerdiumAppData\Partitions` for the portable version) is not deleted.

### Steps to reproduce

1. Create a new Ferdium service.
2. Check in `.\FerdiumAppData\FerdiumAppData\Partitions` : a new folder has been created.
3. Delete the service
4. The partition sub-folder is still there.

**Comment by meiphoo7-Mae** [comment]
Accidently I had to delete a service today because I had the impression that it malfunctioned. After deleting the service the partition sub-folder still existed however after a certain amount of time it did disappear. It's not a matter of seconds but it takes about a minute or so for the sub-folder to disappear. I did not measure the exact amount of time and I observed this on Linux.

**Comment by neel1996** [comment]
Hi, I was analyzing this issue and I can reproduce it on windows 11. I tried to delete the service partition folders from the data directory manually when the electron app is running and found that I was unable to do so. I am able to delete the folders only when the app is entirely shut down.

![image](https://user-images.githubusercontent.com/47709856/196237052-7edf4ade-6be1-42b8-9bbd-312f555fe37f.png)

I am new to electron, so wanted to know if this is an expected behavior.

**Comment by vraravam** [comment]
While any process is running, and if it has file pointers open, then the file-lock mechanism in any OS will prevent it from being deletable. That is expected behavior for the OS.

**Comment by francwalter** [comment]
How can I see which Partitions folder is the one for which service?
I dont find a settings file where I can read my service's name in the folders.
I deleted a service and the Partitions folders are still the same, before I had 15 and after deleting a service I still have 15 in:

`c:\Users\f\AppData\Roaming\Ferdium\Partitions\`

Where is the setting which cryptic foldername is which service?
I am on Windows 10 with the installed (for only me) Ferdium 6.4.1
Thanks

**Comment by vraravam** [comment]
> Where is the setting which cryptic foldername is which service?

This can be found in the database where you service and its internal metadata are stored. They are not exposed directly to end users since we will then have to field many more questions/pleas for help when someone might do something that we cannot replicate locally.

Also, when you delete a service, its information is not deleted on disk. That will have to be a cleanup operation like re-imaging your hard drive - which can effectively remove such cruft. This is SOP for any/all applications in most OSes that I know about.

**Comment by francwalter** [comment]
> > Where is the setting which cryptic foldername is which service?
> 
> This can be found in the database where you service and its internal metadata are stored. 

And where is that?
I searched a while, neither did I find the settings for a service. The only config I found in partitions folder was that Database.db file in e.g.:
`c:\Users\f\AppData\Roaming\Ferdium\Partitions\service-e34f89ec-e7cd-43f5-8676-747738691458\databases\Databases.db`
I opened it with sqlite3 but didnt find interesting things.

Does Ferdium just open each folder in partitions and look for Databases.db files?
Or where is the database for ferdium to see which partitions folders there are?
I want to manually delete the folder, on my tablet I have very few space and need to make more.

**Comment by vraravam** [comment]
If you have signed up using an accountless (or local server) option, then the location of this db file is: `<ProfileFolder>/server.sqlite`. If you have signed up with a hosted server, then that file is not available for you by default. But, you can go to `Help > Import/Export Data` (something like that) and then login into your account via the browser page that opens up. I don't remember if the partition details would show up there though (it might).

**Comment by MurzNN** [comment]
For hosted accounts seems we can find the uuid of the service in cache like `.config/Ferdium/Cache/Cache_Data/1fa7285c9e16ccc3_0`:
```
{
  "customRecipe": false,
  "hasCustomIcon": false,
  "isBadgeEnabled": true,
  "isDarkModeEnabled": false,
  "isEnabled": true,
  "isMuted": false,
  "isNotificationEnabled": true,
  "order": 3,
  "spellcheckerLanguage": null,
  "workspaces": [],
  "recipeId": "msteams",
  "isHibernationEnabled": false,
  "isWakeUpEnabled": true,
  "customIcon": null,
  "userAgentPref": null,
  "name": "Teams",
  "isIndirectMessageBadgeEnabled": false,
  "proxy": {
    "isEnabled": false,
    "host": "",
    "port": 0,
    "user": "",
    "password": ""
  },
  "darkReaderSettings": {
    "brightness": "",
    "contrast": "",
    "sepia": ""
  },
  "isMediaBadgeEnabled": false,
  "trapLinkClicks": false,
  "isProgressbarEnabled": false,
  "iconUrl": null,
  "id": "89949801-e5dd-42af-88d5-70fcfa93cab3",
  "userId": 48
},
```
So, maybe still make the uuid available via Ferdium UI? Cuz really, my Partitions folder size is 27Gb for now!

**Comment by francwalter** [comment]
> ... you can go to `Help > Import/Export Data` (something like that) and then login into your account via the browser page that opens up.

Thank you lottly 😄 
Yes, indeed I use the Ferdium Server to keep my accounts.
And that works. I opened (logged in with my ferdium accounnt) [Your Account Data](https://api.ferdium.org/user/data) and here I can read the uuids.
Now I can delete the apropriate folders, one e.g. having 130 MB of never more used data in it 😁 

I cannot delete the folder totally, somethings still using them, but after a Windows reboot, it was possible.

Strange, that the folders are not deleted when I delete the service. There should be at least a kind of CheckBox "Do you want to delete all files on disk as well?" or so. And the folder should be totally deletable after I delete the service.
As in the issue posted.

**Comment by francwalter** [comment]
> Also, when you delete a service, its information is not deleted on disk. That will have to be a cleanup operation like re-imaging your hard drive - which can effectively remove such cruft. This is SOP for any/all applications in most OSes that I know about.

I have not tried but I assume that an uninstall of whole Ferdium itself will delete all these files. At least this would be default behavior of most uninstall of applications I know. Some apps ask on uninstall if you want to keep settings (and data), but without that switch I assume that nothing of the app should remain, which is the meaning of uninstall/remove. 
To reset Windows (and delete all data) just to clean rests of applications is not a considerable solution for me. I would spend weeks till all apps I "need" would be back in order. 

Thank you for showing how to find the right partitions folder, I hope that this issue will be solved by deleting that folder with delete service.

**Comment by vraravam** [comment]
> I assume that an uninstall of whole Ferdium itself will delete all these files

I dont think this is default behavior in any apps that I use on macos nor on Windows. Snaps/flatpacks/portable apps might behave differently - I am not sure.

**Comment by linfejs** [comment]
hey i just started with ferdium and i cant delete any service properly. I can but every time i restart ferdium the service shows up again. anyone knows whats wrong with it?
